lib/internal/fs/promises.js

```diff
@@ -1181,6 +1181,35 @@ async function mkdtemp(prefix, options) {
   );
 }
 
+async function mkdtempDisposable(prefix, options) {
+  options = getOptions(options);
+
+  prefix = getValidatedPath(prefix, 'prefix');
+  warnOnNonPortableTemplate(prefix);
+
+  const cwd = process.cwd();
+  const path = await PromisePrototypeThen(
+    binding.mkdtemp(prefix, options.encoding, kUsePromises),
+    undefined,
+    handleErrorFromBinding,
+  );
+  // we need to stash the full path in case of process.chdir()
+  const fullPath = pathModule.resolve(cwd, path);
+
+  const remove = () => lazyRimRaf()(fullPath, {
+    maxRetries: 0,
+    recursive: true,
+    retryDelay: 0,
+  });
```

Comment:
I believe the convention is to make these explicitly wrapped/awaited, for stacktracing etc. (#58400 et al.)

```suggestion
  const remove = async () => await lazyRimRaf()(fullPath, {
    maxRetries: 0,
    recursive: true,
    retryDelay: 0,
  });
```

Reply:
For readability I recommend separating out the `lazyRimRaf` call into a separate line...

like..

```
const rmrf = lazyRimRaf();
const remove = async() => rmrf(...);
```